File: app/src/main/java/edgar/yodgorbek/sportnews/database/SourceTypeConverter.java
package edgar.yodgorbek.sportnews.database;

import android.arch.persistence.room.TypeConverter;

import edgar.yodgorbek.sportnews.model.Source;

public class  SourceTypeConverter {
    @TypeConverter
    public static String ConvertSource(Source source){
        return source == null ? null : source.toString();
    }

    @TypeConverter
    public static Source ConvertSource(String source){
        return source == null ? null : new Source(source);
    }
}

File: app/src/main/java/edgar/yodgorbek/yangiliklar/database/SourceTypeConverter.kt
package edgar.yodgorbek.yangiliklar.database

import android.arch.persistence.room.TypeConverter

import edgar.yodgorbek.yangiliklar.model.Source

object SourceTypeConverter {
    @TypeConverter
    fun ConvertSource(source: Source?): String? {
        return source?.toString()
    }

    @TypeConverter
    fun ConvertSource(source: String?): Source? {
        return if (source == null) null else Source(source)
    }
}
